builtin/provisioners/file/resource_provisioner.go

```diff
@@ -79,17 +79,11 @@ func applyFn(ctx context.Context) error {
 }
 
 func validateFn(d *schema.ResourceData) (ws []string, es []error) {
-	numSrc := 0
-	if _, ok := d.GetOk("source"); ok == true {
-		numSrc++
+	if d.Get("source").(string) == "" && d.Get("content").(string) == "" {
```

Comment:
That is not needed, as that will be catched by the `ConflictsWith` configs defined in the schema.